### PR TITLE
Final ux

### PR DESCRIPTION
**Pull request by SpursFan21**
in prog

### CODE REVIEW

File: src/bse202/__init__.py [hunk 1]
```diff
@@ -14,16 +14,25 @@
 
 def init_db():
     with app.app_context():
-        # start fresh
+        # Start fresh by deleting existing data
         delete_db()
```
Comment (d3rpp) [inline src/bse202/__init__.py]: this is redundant, the sql files clear the DB, It won't cause issue (probably) but there's no point in adding this

File: src/bse202/__init__.py [hunk 2]
```diff
@@ -14,16 +14,25 @@
...
-        with app.open_resource("sample_data.sql", mode="rt") as f:
-            db.cursor().executescript(f.read().decode("utf-8"))
+
+        # Execute sample_data.sql
+        with app.open_resource("sample_data.sql", mode="rb") as f:
+            sample_data_content = f.read().decode("utf-8")
+            db.cursor().executescript(sample_data_content)
+
+        # Commit changes after executing sample_data.sql
         db.commit()
```
Comment (d3rpp) [inline src/bse202/__init__.py:34]: ngl this verbosity is a good idea :+1:

File: src/bse202/routes/user/__init__.py [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+from .search import search_games
+from .recommendation import get_specific_games
 
 __all__ = ["user_blueprint"]
+
+# Register the search route under user_blueprint
+user_blueprint.route("/search", methods=["GET"])(search_games)
+
+# Register the recommendation route under user_blueprint
+user_blueprint.route("/recommendation", methods=["GET"])(get_specific_games)
```
Comment (d3rpp) [inline src/bse202/routes/user/__init__.py]: what
why?
no.

File: src/bse202/routes/user/recommendation.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from flask import Blueprint, jsonify
+from ...db import get_db
+
+user_blueprint = Blueprint("user", __name__)
```
Comment (d3rpp) [inline src/bse202/routes/user/recommendation.py]: why?

```py
from .blueprint import user_blueprint
```

This is redundant, please check the code around it to see how to use it my friend

File: src/bse202/routes/user/search.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# user/search.py
+
+from flask import Blueprint, request, jsonify
+from ...db import get_db
+import logging
+
+search_bp = Blueprint("search", __name__, url_prefix="/search")
```
Comment (d3rpp) [inline src/bse202/routes/user/search.py]:
```py

from .blueprint import user_blueprint

@user_blueprint.get("/search")
def search_games():
    # function content
```

please look around for more examples, there are other files where this pattern is used

File: src/bse202/static/media/Game/16.png
Comment (d3rpp) [inline src/bse202/static/media/Game/16.png:1]: Ok I guess it's better, but its squished, realistically at this point I don't care, but like, I though by replacing them you meant getting similar aspect ratio images and cropping them?

Again, not an actual issue, but I'm a bit concerned

File: src/bse202/templates/views/login.html [hunk 6]
```diff
@@ -33,15 +33,29 @@ <h1>Login</h1>
...
                 <div class="secondary-buttons">
-                    <button type="button" class="btn-secondary">Create Account</button>
+                    <button type="button" class="btn-secondary" onclick="gotosignup()">Create Account</button>
                     <button type="button" class="btn-secondary">Forgot Password</button>
                 </div>
             </form>
-            {% if error is defined %}
-                <p>{{ error['message'] }}</p>
+            {% if message %}
+                <p>{{ message }}</p>
```
Comment (d3rpp) [inline src/bse202/templates/views/login.html:41]: will we no longer be using the actual sign-up page?
If so i'll remove the GET handler for it

**Review by d3rpp — CHANGES_REQUESTED** [review]
Please look into the how you're using the blueprints as it doesn't align with the rest of the project.

There were a couple of other things but I forgor :skull:

File: src/bse202/templates/views/login.html (hunk 6 above)
Comment (d3rpp) [reply src/bse202/templates/views/login.html:41]: > [!NOTE]
> sorry, wrong lines, right file though

File: src/bse202/static/media/Game/16.png
Comment (SpursFan21) [reply src/bse202/static/media/Game/16.png:1]: yeah that was halims work

File: src/bse202/templates/views/login.html (hunk 6 above)
Comment (SpursFan21) [reply src/bse202/templates/views/login.html:41]: im not sure that was halims other contribution

File: src/bse202/__init__.py (hunk 1 above)
Comment (SpursFan21) [reply src/bse202/__init__.py]: update made i think

Comment (d3rpp) [reply src/bse202/__init__.py]: Could we remove the `delete_db` function?

**Comment by d3rpp** [comment]
Have we tested that the bigger images don't mess with the layouts?

**Review by d3rpp — APPROVED** [review]
lgtm!

**Pull request merged**